### DISCUSSION TITLE
LAGS when preload

### DISCUSSION THREAD

**Discussion by borodadada** [nuxt/nuxt#17427] [Questions] [answered]
i added preload data on server side

```
<script setup>

if(process.server){
    const res = await Promise.all([
        teamspeak.getFullUserList(),
        teamspeak.getInfoForWeb()
    ])

    store.fullUserList = res[0] 
    store.infoForWeb = res[1] 
}

</script>
```
code
```
    async getFullUserList(){
        const url =  this.url() + '?target=teamspeak&action=getFullUserList'
        const res = await $fetch(url)
        if( res && res.status ){
            return res.data
        }else 
            return false
    },
    async getInfoForWeb(){
        const url =  this.url() + '?target=teamspeak&action=getInfoForWeb'
        const res = await $fetch(url)
        if( res && res.status ){
            return res.data
        }else 
            return false
    },
```

if i update browser very fast use F5 lag begins on the NUXT 3 wtf?

i replace $fetch on setTimeout

```
        return new Promise((resolve, reject)=>{
            setTimeout(() => {
                resolve(true)
            }, 400);
        })
```

all work fine, maybe problem with $fetch or http server
after i replace $fetch on axios and same problem = lags
http server ???

**Comment by Diizzayy** [comment, marked as answer]
@borodadada The issue lies with how you're fetching the data, it's not being hydrated properly. Have a look at [Data fetching with Nuxt3](https://v3.nuxtjs.org/guide/features/data-fetching/) for a better understanding of how this process works.

Below is an example of how your code can be restructured, This will work as you'd expect for the initial response as well for client side navigation.

```vue
<script lang="ts" setup>
const { data } = await useAsyncData('my-data', () => Promise.all([
  teamspeak.getFullUserList(),
  teamspeak.getInfoForWeb()
]))
</script>
```

you can then access the results using `data.value`

**Reply by borodadada** [reply]
thx

**Comment by TheAlexLichter** [comment]
Be aware that Node.js is single-threaded and each SSR-request is blocking. If you send lots of requests in short time to any Node.js application, each request has to wait until the previous one finishes. And because SSR is a CPU-heavy task and also has to wait for API calls like in your example, that takes a few milliseconds, this could be the reason.

But it is hard to guess without the exact reproduction, e.g. on StackBlitz or CodeSandbox 🙈

**Reply by borodadada** [reply]
thx

**Comment by borodadada** [comment]
thx guys, problem with server part, rate limit or etc....
